enumfields/fields.py

```diff
@@ -28,10 +31,47 @@ def __get__(self, obj, type=None):
         return obj.__dict__[self.field.name]
 
     def __set__(self, obj, value):
-        obj.__dict__[self.field.name] = self.field.to_python(value)
-
-
-class EnumFieldMixin:
+        initial_field_name = INITIAL_FIELD_TMP.format(self.field.name)
+        python_value = self.field.to_python(value)
+        obj.__dict__.setdefault(initial_field_name, python_value)
+        obj.__dict__[self.field.name] = python_value
```

Comment:
for mee more cleaner seems to use
```
if not hasattr(obj, initial_value_cache_name):
    setattr(obj, initial_value_cache_name, python_value)
setattr(obj, self.field.name, python_value)
```

but I see that who written enum used dict too. I will take up to you

Reply:
@jsilhan but realy good job. Most of python developers don't understand how descriptor works.

Reply:
I think the original author used it to avoid infinite recursion of `__set__` calls.